lib/pluginmanager/sources/local.rb

```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+require "rubygems/package"
+require "pluginmanager/sources/base"
+require "fileutils"
+
+module LogStash::PluginManager::Sources
+
+  class Local < Base
+
+    def exist?
+      ::File.exist?("#{uri}")
+    end
+
+    def fetch(dest="")
+      base_dir = (dest.empty? ? LogStash::Environment::LOGSTASH_HOME : dest )
+      current_path = ::File.join(base_dir, ::File.basename("#{uri}"))
```

Comment:
Oh well, the base class force URI() on everything.

```
u = URI("stove.io/download.zip")
=> #<URI::Generic:0x28b576a9 URL:stove.io/download.zip>
```
And  it fallback to a generic scheme, interesting.